nim_status/accounts/generator/generator.nim

```diff
@@ -175,37 +203,57 @@ proc reset(self: Generator) =
   # Reset resets the accounts map removing all the accounts from memory.
   self.accounts.clear()
 
-proc storeDerivedAccounts*(self: Generator, accountId: UUID, paths: seq[KeyPath],
-  password: string, dir: string = "", version: int = 3,
-  cryptkind: CryptKind = AES128CTR, kdfkind: KdfKind = PBKDF2,
-  workfactor: int = 0): StoreDerivedAccountsResult =
+proc storeKeyFile*(self: Generator, secretKey: SkSecretKey, password: string,
+  dir: string, version: int = 3, cryptkind: CryptKind = AES128CTR,
+  kdfkind: KdfKind = PBKDF2, workfactor: int = 0): StoreKeyFileResult =
+
+  let
+    address = secretKey.toAddress
+    findKeyFileResult = self.findKeyFile(address, dir)
+
+  if findKeyFileResult.isOk:
+    return StoreKeyFileResult.err fmt"Key file for address {address} already " &
+      "exists"
 
   var workfactorFinal = workfactor
   when not defined(release):
     # reduce the account creation time by a factor of 5 for debug builds only
     if workfactorFinal == 0: workfactorFinal = 100
 
+  let keyFileJsonResult = createKeyFileJson(PrivateKey secretKey, password,
+    version, cryptkind, kdfkind, workfactorFinal)
+
+  if keyFileJsonResult.isErr:
+    return StoreKeyFileResult.err "Error creating key file: " &
+      $keyFileJsonResult.error
+
+  let
+    keyFileJson = $keyFileJsonResult.get
+    now {.used.} = now().format("yyyy-MM-dd'T'HH-mm-ss'.'fffffffff'Z'")
+    keyFilePath = dir / fmt"UTC--{now}--{($address).strip0xPrefix}"
```

Comment:
There's one thing I noticed when trying out the branch (after my earlier PR review/approval):

The format has UTC but `now()` is giving us the local time. For example, a keystore that was created by the example client a few minutes ago has the filename `UTC--2021-07-21T12-24-44.586918000Z--...`, where `12-24-44...` is the local time for me.

We should instead shift the datetime to actually be UTC. I'll figure out how to do it and push a commit.

Reply:
See: https://github.com/status-im/nim-status/pull/229#pullrequestreview-712031157.